sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosContainerIdentity.java

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos.models;
+
+import com.azure.cosmos.implementation.ImplementationBridgeHelpers;
+import com.azure.cosmos.implementation.Paths;
+import com.azure.cosmos.implementation.Strings;
+import com.azure.cosmos.implementation.Utils;
+
+import static com.azure.cosmos.implementation.guava25.base.Preconditions.checkArgument;
+
+/**
+ * Encapsulates the container link associated with the container.
+ * */
+public final class CosmosContainerIdentity {
+
+    private final String databaseName;
+    private final String containerName;
+    private final String containerLink;
+
+    /**
+     * Instantiates a {@link CosmosContainerIdentity} class
+     *
+     * @param databaseName the name of the database
+     * @param containerName the name of the container
+     * */
+    public CosmosContainerIdentity(String databaseName, String containerName) {
+
+        checkArgument(!Strings.isNullOrWhiteSpace(databaseName), "databaseName should not be null nor empty");
+        checkArgument(!Strings.isNullOrWhiteSpace(containerName), "containerName should not be null nor empty");
+
+        this.databaseName = databaseName;
+        this.containerName = containerName;
+        String databaseLink = Utils.joinPath(Paths.DATABASES_ROOT, databaseName);
+        this.containerLink = Utils.joinPath(databaseLink, Paths.COLLECTIONS_PATH_SEGMENT) + containerName;
+    }
+
+    /**
+     * Gets the fully qualified name of the container
+     *
+     * @return the fully qualified name of the container
+     * */
+    public String getContainerLink() {
```

Comment:
why we need to include this in the public API? and also by theory, we do not really need container link here, currently it is only used in the toString() method, we can change to just use database name and container name directly

<img width="1017" alt="image" src="https://user-images.githubusercontent.com/64233642/216428907-f6e439a9-271e-4f0b-a897-3c267e3682cd.png">

Reply:
Agreed - removed